app/models/attachment.rb

```diff
@@ -4,8 +4,6 @@ class Attachment < ActiveRecord::Base
 
   mount_uploader :file_upload, FileUploader
 
-  belongs_to :attachable, polymorphic: true
-
```

Comment:
delete attachments from S3 ?   Do you mean delete those attachments which do not have any references ?